File: src/client/xnative/lion.xios/StoreViewController.cs
using Foundation;
using System;
using System.Net;
using UIKit;

namespace Lion.XiOS
{
    public partial class StoreViewController : UIViewController
    {
        public StoreViewController(IntPtr handle) : base(handle)
        {
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            this.myLodingIndicator.StartAnimating();
            this.PerformSelector(new ObjCRuntime.Selector("LoadMap"), this.myLodingIndicator, 0);
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
        }

        public override void ViewWillAppear(bool animated)
        {
            this.InternetCheck();
        }

        [Export("LoadMap")]
        private void LoadMap()
        {
            var originalString = "로또판매점";

            var escapedString = WebUtility.UrlEncode(originalString);
            var urlString = String.Format("https://m.map.naver.com/search2/search.nhn?query={0}&sm=sug#/map/1", escapedString);

            var theURL = NSUrl.FromString(urlString);
            var urlRequest = NSUrlRequest.FromUrl(theURL);

            this.myWebView.LoadRequest(urlRequest);
            this.myLodingIndicator.StopAnimating();
        }

        private void InternetCheck()
        {
            //if (Reachability.ReachabilityForInternetConnection().CurrentReachabilityStatus() == NotReachable)
            //{
            //    var appDelegate = (AppDelegate)UIApplication.SharedApplication().TheDelegate();
            //    appDelegate.Logined = false;
            //    appDelegate.UserToken = "";
            //    UIAlertController alertController = UIAlertController.AlertControllerWithTitleMessagePreferredStyle("?곌껐 ?ㅻ쪟", "?ㅽ듃?뚰겕 ?곌껐 ?곹깭 ?뺤씤 ??n?ㅼ떆 ?쒕룄?댁＜?몄슂.", UIAlertControllerStyleAlert);
            //    UIAlertAction actionOk = UIAlertAction.ActionWithTitleStyleHandler("?뺤씤", UIAlertActionStyleDefault, null);
            //    alertController.AddAction(actionOk);
            //    var vc = UIApplication.SharedApplication().TheDelegate().Window().RootViewController();
            //    vc.PresentViewControllerAnimatedCompletion(alertController, true, null);
            //}
            //else
            //{
            //}
        }
    }
}

File: src/client/xcode/Lion.iOS/QLottoLion/StoreViewController.m
//
//  StoreViewController.m
//  QLottoLion
//
//  Created by OdinSoft on 2017. 3. 28..
//  Copyright © 2017년 odinsoftware. All rights reserved.
//

#import "StoreViewController.h"

@interface StoreViewController ()

@property (strong, nonatomic) IBOutlet UIWebView *myWebView;

@end

@implementation StoreViewController
@synthesize myLodingIndicator;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [myLodingIndicator startAnimating];
    
    [self performSelectorOnMainThread:@selector(loadMap) withObject:myLodingIndicator waitUntilDone:NO];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)viewWillAppear:(BOOL)animated{
    //internet check
    [self internetCheck];
    
}

-(void)loadMap{
    // korean character makes to query
    NSString *originalString = @"로또판매점";
    NSString *escapedString = [originalString stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLQueryAllowedCharacterSet];
    
    // Your webView code goes here
    NSString *urlString = [NSString stringWithFormat:@"https://m.map.naver.com/search2/search.nhn?query=%@&sm=sug#/map/1", escapedString];
    NSURL *theURL = [NSURL URLWithString:urlString];
    NSURLRequest *urlRequest = [NSURLRequest requestWithURL:theURL];
    [_myWebView loadRequest:urlRequest];
    NSLog(@"MAP MAP MAP");
    
    [myLodingIndicator stopAnimating];

}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

#pragma mark - check internet
- (void) internetCheck{
    
    NSLog(@"online mode check");
    if ([[Reachability reachabilityForInternetConnection]currentReachabilityStatus]==NotReachable)
    {
        //connection unavailable
        NSLog(@"online mode OFF");
        
        AppDelegate * appDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
        appDelegate.logined = NO;
        appDelegate.userToken = @"";
        
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"연결 오류"
                                                                                 message:@"네트워크 연결 상태 확인 후\n다시 시도해주세요."
                                                                          preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction *actionOk = [UIAlertAction actionWithTitle:@"확인"
                                                           style:UIAlertActionStyleDefault
                                                         handler:nil];
        [alertController addAction:actionOk];
        UIViewController *vc = [[[[UIApplication sharedApplication] delegate] window] rootViewController];
        [vc presentViewController:alertController animated:YES completion:nil];
        
    }else{
        NSLog(@"online mode ON");
    }
    
}

@end
